File: clojure/src/aoc2019/day1.clj
(ns aoc2019.day1
  (:gen-class))

(defn fuel
  ""
  [mass]
  (max 0 (- (quot mass 3) 2)))

(defn fuel2
  ""
  [mass]
  (if (= 0 (fuel (fuel mass)))
    (fuel mass)
    (+ (fuel mass) (fuel2 (fuel mass)))))

(defn part1
  ""
  [input]
  (reduce + (map fuel input)))

(defn part2
  ""
  [input]
  (reduce + (map fuel2 input)))

(defn -main
  [input]
  (do
    (println "part 1:" (part1 (map #(Integer/parseInt %) input)))
    (println "part 2:" (part2 (map #(Integer/parseInt %) input)))))

File: php/day1.php
<?php

function fuel1(int $x): int {
    return intdiv($x, 3) - 2;
}

function fuel2(int $x): int {
    $fuel = 0;
    while (true) {
        $f = fuel1($x);
        if ($f <= 0) {
            break;
        }
        $fuel += $f;
        $x = $f;
    }
    return $fuel;
}

$input = file(__DIR__.'/../input/1.txt');
$fuel1 = 0;
$fuel2 = 0;
foreach ($input as $line) {
    $line = intval($line);
    $fuel1 += fuel1($line);
    $fuel2 += fuel2($line);
}

echo "day1 part1: $fuel1\n";
echo "day1 part2: $fuel2\n";